Jenkinsfile

```diff
@@ -33,8 +35,34 @@ node {
 
     if(env.BRANCH_NAME == "master") {
       stage('Publish Gem') {
-        sshagent(['govuk-ci-ssh-key']) {
-          govuk.runRakeTask("publish_gem --trace")
+        def version = sh(
+          script: /ruby -e "puts eval(File.read('${REPOSITORY}.gemspec'), TOPLEVEL_BINDING).version.to_s"/,
+          returnStdout: true
+        ).trim()
+
+        def taggedReleaseExists = sh(
+          script: "git tag | grep v${version}",
+          returnStatus: true
+        ) == 0
+
+        if (taggedReleaseExists) {
+          echo "Version ${version} has already been tagged on Github"
+        } else {
+          echo('Pushing tag')
+          govuk.pushTag(REPOSITORY, env.BRANCH_NAME, 'v' + version)
+        }
+
+        def versionAlreadyPublished = sh(
+          script: "gem list ^${REPOSITORY}\$ --remote --all --quiet | grep ${version}",
```

Comment:
Oh, do you mean drop the `--all` flag and assume we should only be checking the latest version?